### ISSUE TITLE
Extend to_df to obsm and layers

### ISSUE THREAD

**Issue by Hrovatin** [scverse/anndata#775]
Modify to_df so that user can also convert obsm values or other X layers to df.

**Comment by Zethson** [comment]
Layer should already work? https://github.com/scverse/anndata/blob/f563dadad3fcf08d637a5ce1fb42c84d26481f8f/anndata/_core/anndata.py#L1321=

Would you be interested in filing up a PR for obsm? Should not be too difficult...

**Comment by Hrovatin** [comment]
Will make for obsm

The layers is missing from documentation: https://anndata.readthedocs.io/en/refpaths/anndata.AnnData.to_df.html?highlight=to_df#anndata.AnnData.to_df - not sure how that is generated, maybe an old version?

**Comment by Zethson** [comment]
Yeah, the docstring is malformed. You can look at the other working docstrings in the Python file for examples. Would be great if you could also add a quick fix for it (since you'll also need to document support for `.obsm`. Ping me if lost.

**Comment by Hrovatin** [comment]
I think the docstring problem may be that docs are taken from different branch rather than malformed string - when I look at documentation on main it has the layers arg noted down, but when clicking the source code button in docs I get thrown to branch refpaths which does not have layer arg (e.g. within code) nor the doc string.

**Comment by Zethson** [comment]
Ahh, interesting. Thanks for looking into this. Let's resolve this step by step.

**Comment by Hrovatin** [comment]
Anyway, the code should be in #776
To me the doc strings there looks fine.

**Comment by ivirshup** [comment]
I think this has been discussed offline, but I'm not totally convinced of this feature.

At the moment, this is largely adding functionality equivalent to:

```python
pd.DataFrame(adata.obsm[key], index=adata.obs_names)
```

which I think is pretty trivial for a user to do themselves.
